nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -359,6 +362,28 @@ sub GrubFs {
             fi
         ";
     }
+
+    rmtree("$bootPath/theme") or die "cannot clean up theme folder in $bootPath\n" if -e "$bootPath/theme";
+
+    if ($theme) {
+        # Copy theme
+        rcopy($theme, "$bootPath/theme") or die "cannot copy $theme to $bootPath\n";
+        $conf .= "
+            # Sets theme.
+            set theme=" . ($grubBoot->path eq "/" ? "" : $grubBoot->path) . "/theme/theme.txt
+            export theme
+            # Load theme fonts
+         ";
+
+         find( { wanted => sub { # TODO: recursive glob for *.pf2 in theme, rewrite path from $theme/(.*) to $1
```

Comment:
I want to get this hack removed before merge, anyone with good perl knowledge has any idea?

Reply:
Breaks if the .pf2 file is in a subfolder of the theme, since we're referencing it by it's filename, not sub-path

Reply:
> The wanted function takes no arguments but rather does its work through a collection of variables.
> 
>     `$File::Find::dir` is the current directory name,
>     `$_` is the current filename within that directory
>     `$File::Find::name` is the complete pathname to the file.
> 

https://perldoc.perl.org/File/Find.html#The-wanted-function

I believe you should be able to replace `$_` with `$File::Find::name` to fix the issue with files in subfolders.

Reply:
Alternatively, could we instead leave it to the user to set `font` separately from `theme`?

It's not like it's a self-contained theme system anyway, which is why we have this hack.

Having distinct options allows more customization in the end. The user would use `"${pkgs.nixos-grub-theme}/the/font.pf2"` as a value in that instance.

Or am I missing something?

Reply:
The theme specifies the font, we're just loading it

See ${nixos-grub2-theme}/theme.txt

Reply:
Right, [I see what I did there](https://github.com/NixOS/nixpkgs/blob/82a441d3cfcccce66500e40358a241ed9ccb7c47/nixos/modules/installer/cd-dvd/iso-image.nix#L204-L205).

It's been such a long time I don't remember all that :|

I figure then that this `find` is appropriate, given that AFAICT from superficial searches there is no recursive glob. That's the perl way to do it.